services/payments/datastore.go

```diff
@@ -49,7 +49,7 @@ func (q *QLDBDatastore) InsertPaymentState(ctx context.Context, state *paymentLi
 			// For the transaction, check if this transaction has already existed. If not, perform
 			// the insertion.
 			existingPaymentState, err := txn.Execute(
-				"SELECT * FROM transactions WHERE idempotencyKey = ?",
+				"SELECT metadata.id as documentID FROM _ql_committed_transactions WHERE data.idempotencyKey = ?",
```

Comment:
I think `SELECT d_id FROM transactions BY d_id where idempotencyKey = ?` will do the same thing without having to go into the committed table explicitly.

Reply:
I like it!